src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlJsonDbFunctionsTranslator.cs

```diff
@@ -0,0 +1,102 @@
+                // not text, we transform it.
+                .Select(a => a is JsonTraversalExpression traversal ? traversal.WithReturnsText(false) : a)
+                .ToArray();
+
+            if (!args.Any(a => a.TypeMapping is NpgsqlJsonTypeMapping || a is JsonTraversalExpression))
+                throw new InvalidOperationException("The EF JSON methods require a JSON parameter and none was found.");
+            if (args.Any(a => a.TypeMapping is NpgsqlJsonTypeMapping jsonMapping && !jsonMapping.IsJsonb))
+                throw new InvalidOperationException("JSON methods on EF.Functions only support the jsonb type, not json.");
+
+            return method.Name switch
+            {
+                nameof(NpgsqlJsonDbFunctionsExtensions.JsonContains) => new SqlCustomBinaryExpression(
+                    _sqlExpressionFactory.ApplyTypeMapping(args[1], _jsonbTypeMapping),
+                    _sqlExpressionFactory.ApplyDefaultTypeMapping(args[2]),
+                    "@>",
+                    typeof(bool),
+                    _boolTypeMapping),
+
+                nameof(NpgsqlJsonDbFunctionsExtensions.JsonContained) => new SqlCustomBinaryExpression(
+                    _sqlExpressionFactory.ApplyDefaultTypeMapping(args[1]),
+                    _sqlExpressionFactory.ApplyTypeMapping(args[2], _jsonbTypeMapping),
+                    "<@",
+                    typeof(bool),
+                    _boolTypeMapping),
+
+                nameof(NpgsqlJsonDbFunctionsExtensions.JsonExists) => new SqlCustomBinaryExpression(
+                    _sqlExpressionFactory.ApplyTypeMapping(args[1], _jsonbTypeMapping),
+                    _sqlExpressionFactory.ApplyDefaultTypeMapping(args[2]),
+                    "?",
+                    typeof(bool),
+                    _boolTypeMapping),
+
+                nameof(NpgsqlJsonDbFunctionsExtensions.JsonExistAny) => new SqlCustomBinaryExpression(
+                    _sqlExpressionFactory.ApplyTypeMapping(args[1], _jsonbTypeMapping),
+                    _sqlExpressionFactory.ApplyDefaultTypeMapping(args[2]),
+                    "?|",
+                    typeof(bool),
+                    _boolTypeMapping),
+
+                nameof(NpgsqlJsonDbFunctionsExtensions.JsonExistAll) => new SqlCustomBinaryExpression(
+                    _sqlExpressionFactory.ApplyTypeMapping(args[1], _jsonbTypeMapping),
+                    _sqlExpressionFactory.ApplyDefaultTypeMapping(args[2]),
+                    "?&",
+                    typeof(bool),
+                    _boolTypeMapping),
+
+                _ => null
+            };
+
+            static SqlExpression RemoveConvert(SqlExpression e)
+            {
+                while (e != null && e is SqlUnaryExpression unary &&
+                       (unary.OperatorType == ExpressionType.Convert || unary.OperatorType == ExpressionType.Convert))
```

Comment:
Um... Maybe you wanted to check for another type? (:

Reply:
No, I wanted to *really* make sure it was Convert :)